packages/vite/src/node/plugins/asset.ts

```diff
@@ -87,6 +94,9 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
 
       id = id.replace(urlRE, '$1').replace(/[\?&]$/, '')
       const url = await fileToUrl(id, config, this)
+      if (isEmitAssetsWithModule) {
+        return `import img from ${JSON.stringify(url)};export default img;`
```

Comment:
This will apply to stuff other than images. Might want to make it say `asset` instead of `img`.